lib/experimental/class-wp-webfonts-provider.php

```diff
@@ -30,43 +30,103 @@
  * into styles (in a performant way for the provider service
  * it manages).
  *
- * @since 6.0.0
+ * @since X.X.X
  */
 abstract class WP_Webfonts_Provider {
 
+	/**
+	 * The provider's unique ID.
+	 *
+	 * @since X.X.X
+	 *
+	 * @var string
+	 */
+	protected $id = '';
+
 	/**
 	 * Webfonts to be processed.
 	 *
-	 * @since 6.0.0
+	 * @since X.X.X
 	 *
 	 * @var array[]
 	 */
 	protected $webfonts = array();
 
+	/**
+	 * Array of Font-face style tag's attribute(s)
+	 * where the key is the attribute name and the
+	 * value is its value.
+	 *
+	 * @since X.X.X
+	 *
+	 * @var string[]
+	 */
+	protected $style_tag_atts = array();
+
 	/**
 	 * Sets this provider's webfonts property.
 	 *
 	 * The API's Controller passes this provider's webfonts
 	 * for processing here in the provider.
 	 *
-	 * @since 6.0.0
+	 * @since X.X.X
 	 *
 	 * @param array[] $webfonts Registered webfonts.
 	 */
 	public function set_webfonts( array $webfonts ) {
 		$this->webfonts = $webfonts;
 	}
 
+	/**
+	 * Prints the generated styles.
+	 *
+	 * @since X.X.X
+	 */
+	public function print_styles() {
+		printf(
+			$this->get_style_element(),
+			$this->get_css()
+		);
+	}
+
 	/**
 	 * Gets the `@font-face` CSS for the provider's webfonts.
 	 *
 	 * This method is where the provider does it processing to build the
 	 * needed `@font-face` CSS for all of its webfonts. Specifics of how
 	 * this processing is done is contained in each provider.
 	 *
-	 * @since 6.0.0
+	 * @since X.X.X
 	 *
 	 * @return string The `@font-face` CSS.
 	 */
 	abstract public function get_css();
+
+	/**
+	 * Gets the `<style>` element for wrapping the `@font-face` CSS.
+	 *
+	 * @since X.X.X
+	 *
+	 * @return string The style element.
+	 */
+	protected function get_style_element() {
+		$attributes = $this->generate_style_element_attributes();
+
+		return "<style id='wp-webfonts-{$this->id}'{$attributes}>\n%s\n</style>\n";
```

Comment:
Perhaps we could remove the newlines from here?

```suggestion
		return "<style id='wp-webfonts-{$this->id}'{$attributes}>%s</style>";
```

Reply:
This comes from the existing pattern in other dependencies, i.e. `WP_Styles` and `WP_Scripts`. It's here to keep parity and consistency between all of Core's dependencies.